### PR TITLE
Update Gdrive app to Android N

### PR DESCRIPTION
**Pull request by danielpassos**
See: [AGDROID-584](https://issues.jboss.org/browse/AGDROID-584)

### CODE REVIEW

**Comment by danielpassos** [comment]
@secondsun 👀 ?

**Comment by secondsun** [comment]
[build.gradle] needs mavenCentral changed to jcenter in order to compile.  The build tools are in jcenter but not central.  Because Google hates you.

**Comment by danielpassos** [comment]
@secondsun Good catch. Done!

File: GDrive/app/src/main/java/org/jboss/aerogear/android/cookbook/gdrive/MainActivity.java [hunk 1]
```diff
@@ -119,11 +118,14 @@ private void retrieveFiles() {
...
                 .forClass(Files.class);
 
+        @SuppressWarnings("unchecked")
         Pipe<Files> documentsPipe = PipeManager.getPipe("files", this);
         documentsPipe.read(new Callback<List<Files>>() {
             @Override
             public void onSuccess(final List<Files> fileses) {
-                Toast.makeText(getApplicationContext(), getString(R.string.fetched, fileses.size()), Toast.LENGTH_LONG).show();
+                Toast.makeText(getApplicationContext(),
+                        getResources().getQuantityString(R.plurals.fetched, fileses.size()),
```
Comment (jcesarmobile) [inline GDrive/app/src/main/java/org/jboss/aerogear/android/cookbook/gdrive/MainActivity.java]: Should be 
`getResources().getQuantityString(R.plurals.fetched, fileses.size()
, fileses.size())` to replace the %d with the actual number of files

Comment (danielpassos) [inline GDrive/app/src/main/java/org/jboss/aerogear/android/cookbook/gdrive/MainActivity.java]: Done!

**Comment by corinnekrych** [comment]
👀

File: GDrive/README.md [hunk 2]
```diff
@@ -1,67 +1,60 @@
...
-### 1. System Requirements
-
-* [Java 7](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
-* [Gradle 2.1](http://www.gradle.org/)
-* Latest [Android SDK](https://developer.android.com/sdk/index.html) and [Platform version 23](http://developer.android.com/tools/revisions/platforms.html)
+### System Requirements
 
-For the complete instructions about how to setup Google credentials, visit our [OAuth2 documentation guide](https://aerogear.org/docs/guides/security/oauth2-guide/#Google)
+* [Java 8](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+* [Gradle](http://www.gradle.org/)
```
Comment (aidenkeating) [inline GDrive/README.md]: Echoing what @jcesarmobile said [here](https://github.com/aerogear/aerogear-android-cookbook/pull/65/files?diff=split#r86980439). I had to manually update my gradle version to one provided in an error when I tried to initially build.

Comment (danielpassos) [inline GDrive/README.md]: What command you did?

Comment (danielpassos) [inline GDrive/README.md]: Try `./gradlew clean build`. In theory it should download everything needed. That is why we have this file commited

Comment (jcesarmobile) [inline GDrive/README.md]: The README.md says `gradle clean build`, without the w, that's the one I used when testing the Two-factor PR and failed

Comment (aidenkeating) [inline GDrive/README.md]: Sorry, I was running `gradle clean build` (in README) and it was giving me a version error. Running `./gradlew clean build` installed the correct version for me. 

I have no idea whether what I just said makes sense or should happen?

Comment (danielpassos) [inline GDrive/README.md]: Makes sense, that's why I've asked what command you guys are running. Going to fix it for all cookbook apps in a bit

**Comment by aidenkeating** [comment]
@danielpassos Followed the README, apart from the `gradlew` bit everything went smoothly. 

Created an project on Google Dev Console (called `AGDroid Drive`), added the credentials to the `Constants.java` file and everything ran smoothly. Not sure why the travis job is failing though.

I have added links to some screenshots below too.

[GDrive Permissions](https://cloud.githubusercontent.com/assets/8698527/20184401/7422dd1e-a760-11e6-9767-3890ce3a38d7.png)

[GDrive File List](https://cloud.githubusercontent.com/assets/8698527/20184400/74022164-a760-11e6-9d4d-0093a9289d82.png)

**Comment by danielpassos** [comment]
@aidenkeating Ignore travis. There is a incorrect travis setup for this repo

**Comment by danielpassos** [comment]
@aidenkeating README fixed!

**Comment by jcesarmobile** [comment]
LGTM

**Comment by danielpassos** [comment]
@aidenkeating @jcesarmobile

**Comment by danielpassos** [comment]
Landed at https://github.com/aerogear/aerogear-android-cookbook/commit/1931061f4cba4aa45f3da05a1efccb5d8565dd46

**Pull request closed without merging**